metricbeat/helper/prometheus/prometheus.go

```diff
@@ -165,6 +166,14 @@ func (p *prometheus) GetProcessedMetrics(mapping *MetricsMapping) ([]common.MapS
 
 				// value may be a mapstr (for histograms and summaries), do a deep update to avoid smashing existing fields
 				update := common.MapStr{}
+				if valueFloat64, ok := value.(float64); ok {
+					if math.IsNaN(valueFloat64) || math.IsInf(valueFloat64, 1) || math.IsInf(valueFloat64, 0) {
+						// Remove the event if value is NaN or +Inf or -Inf
+						delete(eventsMap, keyLabels.String())
```

Comment:
I think this will remove the whole event for metrics having the same labels, I don't think we want that. If you want to remove the event that could have been created on `getEvent` for this metric, you could instead move `getEvent` after this check, so it is only called if the value is going to be stored.

Reply:
Yes you are right! I did not want to remove all events with the same labels. Thanks for catching this!